lib/audited/auditor.rb

```diff
@@ -216,7 +219,24 @@ def audit_destroy
       def write_audit(attrs)
         attrs[:associated] = send(audit_associated_with) unless audit_associated_with.nil?
         self.audit_comment = nil
-        run_callbacks(:audit)  { audits.create(attrs) } if auditing_enabled
+
+        if auditing_enabled
+          run_callbacks(:audit) {
+            audit = audits.create!(attrs)
+            reduce_audits_if_needed if attrs[:action] != 'create'
+            audit
+          }
+        end
+      end
+
+      def reduce_audits_if_needed
+        if max_audits && (extra_count = audits.count - max_audits) > 0
+          first, *extra, merge_target = audits.limit(extra_count + 1)
+          (extra + [merge_target]).inject(first) { |memo, audit| audit.merge(memo) }
+
+          merge_target.save!
+          audits.where(id: ([first] + extra).map(&:id)).delete_all
+        end
```

Comment:
This function is quite complex as it is, perhaps it would be best to split into two? 
so this one only checks which audits if any need to be combined and sends them to a second function, say `combine_audits` that gets a relation and combines all the audits. that would be beneficial also so users can create rake tasks to bulk combine audits (which could be important when adding this to existing databases, as the first run may be very long) or to allow future capabilities (for example, collapse audits older then a certain date). 
Another possible optimization (which would possibly make the audit merge method unnecessary) would be to use something along the lines of `audits_to_combine.last.audited_changes = audits_to_combine.pluck(:audited_changes).reduce(&:merge)`.
One last comment, when collapsing audits it might be a good idea to add a line to the audit comment indicating this audit is the result of multiple audits being combined or indicate this in some other way.